[Bug] AuthFlowModal.tsx has 3 mismatched closing tags (</form> instead of </div>) causing TSX compile errors

## Bug Description

Three JSX closing tags in `AuthFlowModal.tsx` incorrectly use `</form>` when they should be `</div>`. This causes 3 TypeScript compile errors (`npm run check` fails).

## Errors
client/src/components/AuthFlowModal.tsx(114,7): error TS17002: Expected corresponding JSX closing tag for 'div'
client/src/components/AuthFlowModal.tsx(129,7): error TS17002: Expected corresponding JSX closing tag for 'div'
client/src/components/AuthFlowModal.tsx(532,7): error TS17002: Expected corresponding JSX closing tag for 'div'

## Root Cause

All 3 occurrences are the same bug — `</form>` used where `</div>` is correct:

### 1. `SecurityNotice()` component (line 114)
Opening tag is `<div>` (line 100), closing tag is `</form>` (line 114) — should be `</div>`

### 2. `DevelopmentNotice()` component (line 129)
Opening tag is `<div>` (line 122), closing tag is `</form>` (line 129) — should be `</div>`

### 3. Main `AuthFlowModal` return (line 532)
Opening tag is `<div>` (line 465), closing tag is `</form>` (line 532) — should be `</div>`

## Expected Behavior

- `npm run check` should pass with 0 errors
- All closing tags should match their corresponding opening tags

## Affected File

- `client/src/components/AuthFlowModal.tsx` — lines 114, 129, 532

## Fix

Replace `</form>` with `</div>` on lines 114, 129, and 532.

## Labels

- `bug`
- `good first issue`

--- a/client/src/components/AuthFlowModal.tsx
+++ b/client/src/components/AuthFlowModal.tsx
@@ -111,7 +111,7 @@ function SecurityNotice() {
           </div>
         </div>
       </div>
-    </form>
+    </div>
   );
 }
 
@@ -126,7 +126,7 @@ function DevelopmentNotice() {
           Development Environment: Use local .env.local seeded clinician credentials to bypass or test dashboard integrations.
         </p>
       </div>
-    </form>
+    </div>
   );
 }
 
@@ -529,6 +529,6 @@ export function AuthFlowModal({ initialMode, isOpen, onClose }: AuthFlowModalPro
           </div>
         </section>
       </motion.div>
-    </form>
+    </div>
   );
 }

